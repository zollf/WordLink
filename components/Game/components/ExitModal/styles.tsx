import { StyleSheet } from "react-native";
import { widthPercentageToDP, heightPercentageToDP } from 'react-native-responsive-screen';

import AppStyle from "../../../../styles";

const styles = StyleSheet.create({
  wrapper: {
    height: "100%",
    width: "100%",
    position: "absolute",
    display: "flex",
    justifyContent: "center",
    alignItems: "center",
    zIndex: 3,
  },
  exitModal: {
    width: widthPercentageToDP('60%'),
    backgroundColor: AppStyle.lightGrey,
    paddingBottom: widthPercentageToDP('4%'),
    paddingTop: heightPercentageToDP('5%'),
    borderRadius: 5,
    marginBottom: 40,
    display: "flex",
    justifyContent: "center",
    alignItems: "center",
  },
  backing: {
    left: 0,
    height: "100%",
    width: "100%",
    position: "absolute",
    backgroundColor: "black",
    opacity: 0.5,
    zIndex: 2,
  },
  button: {
    marginBottom: 10,
    marginTop: 10,
  },
  text: {
    width: widthPercentageToDP('50%'),
    textAlign: "center",
    color: 'white',
    fontFamily: AppStyle.fontBold,
    fontSize: heightPercentageToDP('3%'),
    marginBottom: 5,
  },
  quit: {
    left: 10,
    top: 10,
    position: "absolute",
    height: "100%",
    width: "100%",
  }
});

export default styles;