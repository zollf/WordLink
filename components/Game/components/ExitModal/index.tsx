import React from 'react';
import { View, Text, LayoutAnimation } from 'react-native';
import { LinearGradient } from 'expo-linear-gradient';
import Button from '../../../Button';
import CircleButton from '../../../CircleButton';

import AppStyle from '../../../../styles';
import styles from './styles';

interface Props {
  close: () => void;
  exit: () => void;
}

const ExitModal = ({ close, exit }: Props) => {
  LayoutAnimation.configureNext({
    duration: 200,
    update: { type: 'spring', springDamping: 0.9 },
  });
  return (
    <>
      <View style={styles.wrapper}>
        <LinearGradient
          colors={[AppStyle.orange, AppStyle.blue]}
          style={styles.exitModal}
          locations={[0.5, 0.5]}
          start={[-0.1, 0.6]}
        >
          <View style={styles.quit}>
            <CircleButton overrideCallback={close} />
          </View>
          <Text style={styles.text}>Are you sure?</Text>
          <View style={styles.button}>
            <Button onPress={exit} text="QUIT" />
          </View>
        </LinearGradient>
      </View>
      <View style={styles.backing} />
    </>
  );
};

export default ExitModal;
