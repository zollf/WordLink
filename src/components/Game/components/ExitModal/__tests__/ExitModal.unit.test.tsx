import React from 'react';
import useWithStores from 'app/test/useWithStores';

import ExitModal from '..';

import renderer from 'react-test-renderer';

const close = jest.fn();
const exit = jest.fn();

const component = useWithStores(<ExitModal close={close} exit={exit} />);

describe('<ExitModal />', () => {
  let wrapper: renderer.ReactTestRenderer;
  beforeEach(() => {
    wrapper = renderer.create(component);
  });

  it('matches snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('close buttons works', () => {
    wrapper.root.findByProps({ 'data-test-id': 'circle-button' }).props.onPress();
    expect(close).toBeCalled();
  });

  it('set current page to menu', () => {
    wrapper.root.findByProps({ 'data-test-id': 'exit-button' }).props.onPress();
    expect(exit).toBeCalled();
  });
});
