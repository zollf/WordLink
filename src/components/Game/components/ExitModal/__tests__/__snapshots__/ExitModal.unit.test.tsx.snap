// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ExitModal /> matches snapshot 1`] = `
Array [
  <View
    style={
      Object {
        "alignItems": "center",
        "display": "flex",
        "height": "100%",
        "justifyContent": "center",
        "position": "absolute",
        "width": "100%",
        "zIndex": 3,
      }
    }
  >
    <View
      style={
        Array [
          undefined,
          Object {
            "opacity": mockConstructor {},
            "transform": Array [
              Object {
                "translateX": mockConstructor {},
              },
              Object {
                "scale": 1,
              },
            ],
          },
        ]
      }
    >
      <ViewManagerAdapter_ExpoLinearGradient
        proxiedProperties={
          Object {
            "colors": Array [
              4294950464,
              4282432255,
            ],
            "endPoint": undefined,
            "locations": Array [
              0.5,
              0.5,
            ],
            "startPoint": Array [
              -0.1,
              0.6,
            ],
          }
        }
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#F3F3F3",
            "borderRadius": 5,
            "display": "flex",
            "justifyContent": "center",
            "marginBottom": 40,
            "paddingBottom": 30,
            "paddingTop": 66.5,
            "width": 450,
          }
        }
      >
        <View
          style={
            Object {
              "height": "100%",
              "left": 10,
              "position": "absolute",
              "top": 10,
              "width": "100%",
            }
          }
        >
          <View
            accessible={true}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "opacity": 1,
              }
            }
          >
            <ViewManagerAdapter_ExpoLinearGradient
              proxiedProperties={
                Object {
                  "colors": Array [
                    4284860671,
                    4282432255,
                  ],
                  "endPoint": undefined,
                  "locations": Array [
                    0.5,
                    0.5,
                  ],
                  "startPoint": Array [
                    -0.1,
                    0.1,
                  ],
                }
              }
              style={
                Object {
                  "alignItems": "center",
                  "borderRadius": 30,
                  "display": "flex",
                  "height": 60,
                  "justifyContent": "center",
                  "shadowColor": "#000",
                  "shadowOffset": Object {
                    "height": 2,
                    "width": 2,
                  },
                  "shadowOpacity": 0.1,
                  "shadowRadius": 4,
                  "width": 60,
                }
              }
            >
              <Image
                source={1}
                style={
                  Object {
                    "height": 30,
                    "width": 30,
                  }
                }
              />
            </ViewManagerAdapter_ExpoLinearGradient>
          </View>
        </View>
        <Text
          style={
            Object {
              "color": "white",
              "fontFamily": "Raleway_700Bold",
              "fontSize": 40,
              "marginBottom": 5,
              "textAlign": "center",
              "width": 375,
            }
          }
        >
          Are you sure?
        </Text>
        <View
          style={
            Object {
              "marginBottom": 10,
              "marginTop": 10,
            }
          }
        >
          <View
            style={
              Object {
                "transform": Array [
                  Object {
                    "scale": mockConstructor {},
                  },
                ],
              }
            }
          >
            <View
              accessible={true}
              focusable={true}
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "opacity": 1,
                }
              }
            >
              <ViewManagerAdapter_ExpoLinearGradient
                proxiedProperties={
                  Object {
                    "colors": Array [
                      4294956160,
                      4294950464,
                    ],
                    "endPoint": undefined,
                    "locations": Array [
                      0.5,
                      0.5,
                    ],
                    "startPoint": Array [
                      -0.1,
                      0.1,
                    ],
                  }
                }
                style={
                  Object {
                    "alignItems": "center",
                    "borderRadius": 5,
                    "display": "flex",
                    "height": 106.5,
                    "justifyContent": "center",
                    "shadowColor": "#000",
                    "shadowOffset": Object {
                      "height": 2,
                      "width": 2,
                    },
                    "shadowOpacity": 0.1,
                    "shadowRadius": 4,
                    "width": 300,
                  }
                }
              >
                <Text
                  style={
                    Object {
                      "color": "white",
                      "fontFamily": "Raleway_700Bold",
                      "fontSize": 33.5,
                    }
                  }
                >
                  QUIT
                </Text>
              </ViewManagerAdapter_ExpoLinearGradient>
            </View>
          </View>
        </View>
      </ViewManagerAdapter_ExpoLinearGradient>
    </View>
  </View>,
  <View
    style={
      Object {
        "backgroundColor": "black",
        "height": "100%",
        "left": 0,
        "opacity": 0.5,
        "position": "absolute",
        "width": "100%",
        "zIndex": 2,
      }
    }
  />,
]
`;
